### PR TITLE
Drude1D model including docs and tests

### LINKED ISSUE #9071: Add a Drude1D model
A drude model is often used in dust grain extinction (2175 A) and emission (aromatic/PAH features).  Would be very useful to have.

https://en.wikipedia.org/wiki/Drude_model

### PR DESCRIPTION
**Pull request by karllark**
### Description

This pull request adds the Drude1D model to the modeling module.  Docs and tests are included.
Some minor edits to make things cleaner to BlackBody code/tests are included.

Fixes #9071

### CODE REVIEW

File: astropy/modeling/physical_models.py [hunk 1]
```diff
@@ -194,3 +196,125 @@ def lambda_max(self):
...
+            / ((x / x_0 - x_0 / x) ** 2 + (fwhm / x_0) ** 2)
+        )
+
+    @staticmethod
+    def fit_deriv(x, amplitude, x_0, fwhm):
+        """
+        Drude1D model function derivatives.
+        """
+        d_amplitude = (fwhm / x_0) ** 2 / ((x / x_0 - x_0 / x) ** 2 + (fwhm / x_0) ** 2)
+        d_x_0 = (
```
Comment (bsipocz) [inline astropy/modeling/physical_models.py]: maybe it's just me, but this formatting is super difficult to read.

Comment (karllark) [reply astropy/modeling/physical_models.py]: I can change this, but suggestions would be needed.  I just use the black program to cleanup all the formatting so I don't have to worry about flake8 errors.

Comment (bsipocz) [reply astropy/modeling/physical_models.py]: yes, I recognize black from miles away now, not sure why they do this, it's hurting ones eyes and adds nothing for readability. I've added it to the upcoming meeting's agenda to come up with a solution that would work for contributors and codebase as well.

Comment (karllark) [reply astropy/modeling/physical_models.py]: That would be great.  There are just too many flake8 rules for me to by hand keep up.  I don't agree with all the choices black makes, but I sure love not having to worry about flake8 stuff anymore!

Comment (bsipocz) [reply astropy/modeling/physical_models.py]: autopep8 is another cmd line tool, it usually is more conservative than black. And almost all editors have plugins that notify you on the spot, but that indeed takes some time to get used to.

**Pull request merged**